src/hotspot/share/runtime/mutex.hpp

```diff
@@ -81,8 +82,14 @@ class Mutex : public CHeapObj<mtSynchronizer> {
        native         = max_nonleaf    +   1
   };
 
+ private:
+  // The _owner field is only set by the current thread, either to itself after it has acquired
+  // the low-level _lock, or to NULL after it has released the _lock. Accesses by any thread other
+  // than the lock owner are inherently racy.
+  Thread* _owner;
```

Comment:
My understanding of the idiom of using Atomic always rather than relying on any semantics for volatile is that the volatile qualifier is left on the variable, as a marker/indicator/clue that it is interesting.  This will also help us find such if/when we adopt some kind of atomic class for encapsulation (whether std::atomic or something like what was proposed for JDK-8247213) and help track adoption progress.  So I think the volatile qualifier should be retained.

Reply:
Yes, sorry.  I thought we'd decided what Kim said.